### PR TITLE
Issue 21882 osgi upload share lock

### LINKED ISSUE #21882: OSGI restart needs to be cluster aware
When uploading an OSGI plugin, only one server in a cluster should read the uploaded plugin, move it to `felix/load` and re-write the `osgi-extras.conf`.   After that is done, when we restart the osgi framework, we need to restart it on all the servers in the cluster so they pick up the changes in `osgi-extras.conf`.

Additionally, when a user clicks the "restart framework" button, it should send a cluster message and restart it across all nodes in the cluster.

### PR DESCRIPTION
**Pull request by jdotcms**
Implementing a cluster share lock, this allows a block cluster wide over parallel threads running on diff nodes (by using a db table by default)

### CODE REVIEW

File: dotCMS/dependencies.gradle [hunk 1]
```diff
@@ -4,6 +4,7 @@ dependencies {
     def eeType = dotcmsReleaseVersion.equals("master") ? '-SNAPSHOT' : ''
 
     compile group: 'com.dotcms.enterprise', name: 'ee', version: dotcmsReleaseVersion + eeType, changing: true
+    //compile fileTree(dir: 'src/main/enterprise/build/libs', include: ['ee_clean.jar'])
```
Comment (nollymar) [inline dotCMS/dependencies.gradle:7]: Please remove this line

**Comment by nollymar** [comment]
MSSQL tests aren't running: https://raw.githack.com/dotCMS/test-results/issue-21882-osgi-upload-share-lock/projects/core/current/integration/mssql/reports/html/packages/com.dotcms.html

**Review by nollymar — CHANGES_REQUESTED** [review]
Please check out MSSQL tests

**Comment by jdotcms** [comment]
> 

it seems to be the same thing that Fabrizzio is looking

**Comment by jdotcms** [comment]
> MSSQL tests aren't running: https://raw.githack.com/dotCMS/test-results/issue-21882-osgi-upload-share-lock/projects/core/current/integration/mssql/reports/html/packages/com.dotcms.html

I have add a fix in order to avoid start the sub pub on mssql, since it only runs on postgres

**Pull request merged**